app/src/main/java/org/stacks/app/ui/auth/signup/ChooseUsernameViewModel.kt

```diff
@@ -0,0 +1,80 @@
+package org.stacks.app.ui.auth.signup
+
+import androidx.hilt.lifecycle.ViewModelInject
+import kotlinx.coroutines.channels.BroadcastChannel
+import kotlinx.coroutines.flow.*
+import org.blockstack.android.sdk.toBtcAddress
+import org.stacks.app.data.ProfileModel
+import org.stacks.app.domain.*
+import org.stacks.app.domain.CheckUsernameStatus.UsernameStatus.Available
+import org.stacks.app.ui.BaseViewModel
+import timber.log.Timber
+
+class ChooseUsernameViewModel
+@ViewModelInject constructor(
+    checkUsernameStatus: CheckUsernameStatus,
+    generateNewIdentityKeys: GenerateNewIdentityKeys,
+    generateIdentity: GenerateIdentity,
+    uploadProfile: UploadProfile,
+    registrarProfile: RegistrarProfile,
+    uploadWallet: UploadWallet
+) : BaseViewModel() {
+
+    // Inputs
+    private val usernameSubmitted = MutableStateFlow("")
+    private val usernameValidated = BroadcastChannel<Unit>(1)
+
+    // Outputs
+    private val newAccount = BroadcastChannel<Unit>(1)
```

Comment:
Not a great name, what does this do? Tell the view what should happen. Is it `openNewAccountScreen`?